src/main/java/com/amazonaws/services/kinesis/leases/impl/KinesisLeaseSelector.java

```diff
@@ -0,0 +1,43 @@
+package com.amazonaws.services.kinesis.leases.impl;
+
+import com.amazonaws.services.kinesis.leases.interfaces.ILeaseSelector;
+
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * KinesisLeaseSelector abstracts away the lease selection logic from the application code that's using leasing.
+ * It owns filtering of the leases to be taken.
+ */
+public class KinesisLeaseSelector<T extends Lease> implements ILeaseSelector<T> {
```

Comment:
Doesn't need to be public. At the creation of the `LeaseCoordinator` if `LeaseSelector` is passed use the one that is passed otherwise create this as the default one.